src/app/upgrade.cpp

```diff
@@ -75,10 +85,66 @@ namespace
             , QLatin1String("Preferences/WebUI/HTTPS/KeyPath")
             , Utils::Fs::toNativePath(configPath + QLatin1String("WebUIPrivateKey.pem")));
     }
+
+#if defined Q_OS_WIN
+    void upgradeResumeFiles()
+    {
+        QString backupFolderPath = Utils::Fs::expandPathAbs(specialFolderLocation(SpecialFolder::Data) + "BT_backup");
+        QDir backupFolderDir(backupFolderPath);        
+
+        const QStringList backupFiles = backupFolderDir.entryList(QStringList(QLatin1String("*.fastresume")), QDir::Files, QDir::Unsorted);
+        const QRegularExpression rx(QLatin1String("^([A-Fa-f0-9]{40})\\.fastresume$"));
+        for (const QString &backupFile : backupFiles) {
+            const QRegularExpressionMatch rxMatch = rx.match(backupFile);
+            if (rxMatch.hasMatch()) {
+                const QString hashStr = rxMatch.captured(1);
+                const QString filepath = backupFolderDir.absoluteFilePath(backupFile);
+
+                QFile file1(filepath);
+                if (!file1.open(QIODevice::ReadOnly))
+                    continue;
+
+                QByteArray data = file1.readAll();
+                file1.close();
+
+                libtorrent::error_code ec;
+                libtorrent::bdecode_node fastOld;
+                libtorrent::bdecode(data.constData(), data.constData() + data.size(), fastOld, ec);
+                if (ec || (fastOld.type() != libtorrent::bdecode_node::dict_t))
+                    continue;
+
+                libtorrent::entry fastNew;
+                fastNew = fastOld;
+
+                const QRegularExpression rx_path(QLatin1String("^.+(//.+)$"));
+                const QRegularExpressionMatch rxMatchPath = rx_path.match(QString::fromStdString(fastNew["save_path"].string()));
+                if (rxMatchPath.hasMatch()) {
+                    const QString old_save_path = QString::fromStdString(fastNew["save_path"].string());
+                    const QString replace_path = rxMatchPath.captured(1).replace('/', '\\');
+                    fastNew["save_path"] = replace_path.toStdString();
+                    QVector<char> out;
```

Comment:
you should reserve `out`: `out.reserve(fastNew.size() * n);` where `n` is efficiency factor of bencoding. Since I didn't find any efficiency factor of bencoding you might go with `1.2` or some other approximate value.

Reply:
It's overthinking... fixed data size will be less or equal to original one.

Reply:
ah yes, I didn't consider the context, then it can be the same as the original one.